core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -729,21 +728,10 @@ public static Exists createExistsPlan(
     }
 
     // for IN/NOT IN, it needs to output the fields
-    final List<RexNode> exprs = new ArrayList<>();
-    if (subQueryType == SubQueryType.IN) {
-      for (int i = 0; i < keyCount; i++) {
-        exprs.add(rexBuilder.makeInputRef(ret, i));
-      }
-    }
-
-    final int projectedKeyCount = exprs.size();
-    exprs.add(rexBuilder.makeLiteral(true));
-
     ret = relBuilder.push(ret)
-        .project(exprs)
         .aggregate(
-            relBuilder.groupKey(ImmutableBitSet.range(projectedKeyCount)),
-            relBuilder.min(relBuilder.field(projectedKeyCount)))
+            relBuilder.groupKey(relBuilder.fields()),
+            relBuilder.literalAgg(true))
```

Comment:
what does LITERAL_AGG return for an empty set?

Reply:
`select literal_agg(true) from emp where 1=0;`
I make this work in Calcite, and it returns 0 rows.